SPHINXsys/src/shared/particles/base_particles.h

```diff
@@ -160,7 +160,7 @@ namespace SPH
 			std::cout << "\n Error: the old variable '" << old_variable_name << "' is not registered!" << std::endl;
 			std::cout << __FILE__ << ':' << __LINE__ << std::endl;
 			exit(1);
```

Comment:
you have exit return is not reachable. print string and use exit? why not throw exception directly?

Reply:
I do not know how to  throw exception in proper way, and will try to revise all such places after learned how to do it. 
They will be updated in the next pull request.